sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -877,21 +877,27 @@ class SessionCatalog(
     isTemporaryTable(nameParts.asTableIdentifier)
   }
 
+  private def lookupTempView(name: TableIdentifier): Option[LogicalPlan] = {
+    val table = formatTableName(name.table)
+    val dbName = formatDatabaseName(name.database.getOrElse(currentDb))
```

Comment:
we don't need to check the current database, as we can't set global_tmp as the current database.

Reply:
The previous code didn't check the current database as well.

Reply:
@cloud-fan This is "previous" code: https://github.com/apache/spark/blob/e79c1cde1bb6751174bfbe4d201f97d700ac0d90/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L998
or you are speaking about another previous code?

Reply:
I mean https://github.com/apache/spark/pull/31265/files#diff-9dd0899e5406230aeff96654432da54f35255f6dc60eecb87264a5c508a8c826L890

Reply:
Sorry, do you propose to call `database.get`?

Reply:
yes, let's follow that "previous" code.

Reply:
I did that. ... The function name `isTemporaryTable` looks weird, so, I would replace it by:
```scala
def isTempView(name: TableIdentifier): Boolean
```

Reply:
Here is the PR https://github.com/apache/spark/pull/31295 which renames `isTemporaryTable()`.